### ISSUE TITLE
Verify Sigstore signatures of CPython artifacts in CI

### ISSUE THREAD

**Issue by sethmlarson** [pyenv/pyenv#3087]
Sigstore is capable of verifying SHA256 checksums against a known release manager identity. Because pyenv is a common method of installing Python on machines, my proposal is to check Sigstore bundles against the SHA256 checksums so pull requests adding new CPython versions are automatically verified.

By putting the verification step in CI, this means users running `pyenv install` won't need to do the verification themselves, instead deferring to pyenv's verification.

I see that there isn't any Python being run in the current CI, if this idea is acceptable to maintainers I'd like some guidance on approach so I can work on the implementation.

**Comment by native-api** [comment]
[We already do checksum verification](https://github.com/pyenv/pyenv/tree/master/plugins/python-build#checksum-verification). Unless I'm misunderstanding you.

**Comment by sethmlarson** [comment]
@native-api I'm aware you do checksum verification, I am suggesting adding verification of those checksums against the expected release manager identity using Sigstore in CI:

```sh
$ wget https://www.python.org/ftp/python/3.13.0/Python-3.13.0.tar.xz.sigstore
$ sigstore verify identity \
    --cert-oidc-issuer https://accounts.google.com \
    --cert-identity thomas@python.org \
    --bundle Python-3.13.0.tar.xz.sigstore \
    sha256:086de5882e3cb310d4dca48457522e2e48018ecd43da9cdf827f6a0759efb07d

OK: SHA2_256:086de5882e3cb310d4dca48457522e2e48018ecd43da9cdf827f6a0759efb07d
```

This would mean maintainers don't need to manually verify checksums themselves when submitted by contributors and protect pyenv from using poisoned artifacts if python.org/downloads were ever compromised.

**Comment by native-api** [comment]
> verification of those checksums against the expected release manager identity

So we'll have to somehow track this "realease manager identity"? Because I surmise it's going to change with some frequency.

We'll also need to somehow get sigstore URLs. So far, download URLs are internal Python-Build logic.

**Comment by sethmlarson** [comment]
Hello, I've discovered a new method you may be interested in after talking more to Sigstore folks: [Cosign](https://github.com/sigstore/cosign) supports offline verification and it's a pre-compiled binary. Please take a look: https://discuss.python.org/t/pep-761-deprecating-pgp-signatures-for-cpython-artifacts/67180/30

**Comment by sethmlarson** [comment]
> So we'll have to somehow track this "release manager identity"? Because I surmise it's going to change with some frequency.

Indeed, but this changes at the same rate as new PGP keys change (once every 2 years, approximately).
